popup.js

```diff
@@ -1,18 +1,119 @@
-document.addEventListener('DOMContentLoaded', function() {
-    console.log('dom content loaded');
-    console.log(chrome);
-    console.log(chrome.tabs);
-    chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
-        console.log(tabs);
-    });
+function setup() {
+    document.addEventListener('DOMContentLoaded', async function() {
+        // fetch language data from the API and dynamically load elements into the drop-down menus.
+        const resp = await fetch('https://libretranslate.eownerdead.dedyn.io/languages');
+        const languageData =  await resp.json();
+        loadLanguages(languageData);
+
+        // setup local storage for the extension to access language features.
+        /*
+        1. Initialise the user's language preferences during translation.
+        2. Set the selected property of the drop-down menus' options accordingly.
+         */
+        const prefs = await initialiseLanguageSelection();
+        console.log('What is prefs? ' + JSON.stringify(prefs));
+        initialiseLanguagePreferences(prefs.source, prefs.target);
 
-    const downloadButton = document.querySelector('button');
-    if (downloadButton) {
-        downloadButton.addEventListener('click', function() {
-            setupDownloadButton();
+        sourceLanguageListener();
+        targetLanguageListener();
+
+        console.log('dom content loaded');
+        console.log(chrome);
+        console.log(chrome.tabs);
+        chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
+            console.log(tabs);
         });
+
+        const downloadButton = document.querySelector('button');
+        if (downloadButton) {
+            downloadButton.addEventListener('click', function() {
+                setupDownloadButton();
+            });
+        }
+    });
+}
+
+/*
+Function loadLanguages collects a list of languages and appends them to both
+drop-down menus.
+ */
+function loadLanguages(languages) {
+    const source = document.querySelector('#source-lang');
+    const target = document.querySelector('#target-lang');
+    for (const language of languages) {
+        const sourceOption = document.createElement('option');
+        sourceOption.value = language.code;
+        sourceOption.text = language.name + '.';
```

Comment:
why the full stop??

Reply:
Removed the unnecessary full-stop, language options will appear as-is in the drop-down menus.